### ISSUE TITLE
📝 [Proposal]: v3 auto binding 

### ISSUE THREAD

**Issue by febelery** [gofiber/fiber#3005]
### Feature Proposal Description

I suggest binding to add an auto-bind method, like the golang echo framework's bind, to automatically parse based on content-type, Form, Body, JSON methods are too cumbersome

### Alignment with Express API

N/a

### HTTP RFC Standards Compliance

N/a

### API Stability

N/a

### Feature Examples
v2 code like: 

```go

type answerReq struct {
	UserId   int `json:"-" query:"-" params:"-"`
	QuizId   int `params:"id" json:"-" validate:"required"`
	RecordId int `json:"id" validate:"required"`
	Duration int        `json:"duration" validate:"required,min=3"`
	Answer   [][]string `json:"answer" validate:"required"`
}

func RequestParser[T any](c *fiber.Ctx) (*T, error) {
	req := new(T)

	if strings.ToLower(c.Method()) != "get" {
		if err := c.BodyParser(req); err != nil {
			if errors.Is(err, fiber.ErrUnprocessableEntity) {
				return nil, errors.New("errors")
			}
			return nil, err
		}
	}
	if err := c.QueryParser(req); err != nil {
		return nil, err
	}
	if err := c.ParamsParser(req); err != nil {
		return nil, err
	}

	if err := validator.Validate(req); err != nil {
		return nil, err
	}

	return req, nil
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have searched for existing issues that describe my proposal before opening this one.
- [X] I understand that a proposal that does not meet these guidelines may be closed without explanation.

**Comment by efectn** [comment]
@febelery the feature you requested already included in #2006 PR

**Comment by febelery** [comment]
thx
